src/window/gui/Gui.cpp

```diff
@@ -93,6 +93,16 @@ Gui::Gui() : Gui(std::vector<std::shared_ptr<GuiWindow>>()) {
 
 Gui::~Gui() {
     SPDLOG_TRACE("destruct gui");
+    if (mImGuiIo != nullptr) {
+        if (mImGuiIo->IniFilename != nullptr) {
+            delete[] mImGuiIo->IniFilename;
+        }
+        if (mImGuiIo->LogFilename != nullptr) {
+            delete[] mImGuiIo->LogFilename;
+        }
+        mImGuiIo->IniFilename = nullptr;
+        mImGuiIo->LogFilename = nullptr;
+    }
```

Comment:
do we need to do this manually, does [`ImGui::Shutdown`](https://github.com/ocornut/imgui/blob/5f0acadf7db1b6d469f0771284e2e3f7818443ce/imgui.cpp#L4185) not handle it?

Reply:
not it's allocate with new here https://github.com/Kenix3/libultraship/blob/7ad23ad94ded59952cc586ad66bf05f4699cb94a/src/window/gui/Gui.cpp#L127-L128
so it's doesn't free if we don't say so.

Reply:
That looks awful, just do an strdup

Reply:
Coco didn't change that code. They're just fixing a memory leak.

Reply:
Don't love the fact we aren't using strings here, but that's out of scope